### DISCUSSION TITLE
Make Device for a Module optional

### DISCUSSION THREAD

**Discussion by sessl3r** [netbox-community/netbox#18158] [Q&A]
Hi all,
in our Lab-Setup we document with Netbox we have the situation that many Modules are not always in use but are stored in a shelf for usage. However we must track information especially history of such Modules and can not just delete them.
We already have the patches from [here](https://github.com/netbox-community/netbox/compare/develop...francesco-noc:netbox:13779-change-device-to-module) in place to be able to move Modules.
What is now missing is to remove the assignment of a Module to a Device/ModuleBay without deleting the Module.
Do you think this is possible without too many other issues we would raise with such a change?

The current workaround idea is to create some Devices for storage.

**Comment by Domoninic** [comment]
Perhaps the[ netbox-inventory](https://github.com/ArnesSI/netbox-inventory) plugin would suit your needs here.
It allows you to create Assets to represent Devices and Modules and then assign them to netbox devices or storage.

**Reply by sessl3r** [reply]
Thanks for the idea. This seemed first to be what we think. However only for Devices. Modules must always be recreated when not placed into a Device. Any other ideas?
Or might it be there is some other tool which anyone is aware of would match our needs better.
